src/plan_expand_hypertable.c

```diff
@@ -359,6 +495,14 @@ ts_plan_expand_hypertable_chunks(Hypertable *ht, PlannerInfo *root, Oid parent_o
 	root->simple_rte_array =
 		repalloc(root->simple_rte_array, root->simple_rel_array_size * sizeof(RangeTblEntry *));
 
+#if !(PG96 || PG10)
+	/* Adding partition info will make PostgreSQL consider the inheritance
+	 * children as part of a partitioned relation. This will enable
+	 * partitionwise aggregation. */
+	rel->nparts = list_length(inh_oids);
```

Comment:
why isn't this line inside `build_hypertable_partition_info`

Reply:
fixed